internal/utils/flags/flags.go

```diff
@@ -243,3 +281,56 @@ func (f StringSliceFlag) Register(fs *pflag.FlagSet) {
 func MarkHidden(fs *pflag.FlagSet, name string) {
 	fs.MarkHidden(name) //nolint: errcheck
 }
+
+var (
+	forwardedNames = make(map[string]string)
+)
+
+// DeprecationHandler is a pflag normalizing function meant to handle deprecating and forwarding flags to new names
+func DeprecationHandler(f *pflag.FlagSet, name string) pflag.NormalizedName {
+	newName, ok := forwardedNames[name]
+	if ok {
+		name = newName
+		f.Lookup(name).Name = newName
+	}
+
+	delete(forwardedNames, name)
+
+	return pflag.NormalizedName(name)
+}
+
+// Deprecator is capable of deprecating a flag
+type Deprecator interface {
+	Deprecate(fs *pflag.FlagSet, name string)
+}
+
+// Deprecated has a flag's deprecation information
+type Deprecated struct {
+	Message string
+}
+
+// Deprecate deprecates the deprecated flag
+func (d Deprecated) Deprecate(fs *pflag.FlagSet, name string) {
+	message := d.Message
+	if message == "" {
+		message = fmt.Sprintf("The field '%s' has been marked for deprecation", name)
+	}
+
+	fs.MarkDeprecated(name, message) //nolint: errcheck
+}
+
+// Forwarded is capable of forwarding a flag that needs to be deprecated
+type Forwarded struct {
+	Message string
+	To      string
+}
+
+// Deprecate deprecates the deprecated flag and forwards it to a new name
+func (f Forwarded) Deprecate(fs *pflag.FlagSet, name string) {
+	if f.Message == "" {
+		f.Message = fmt.Sprintf("The field '%s' has been marked for deprecation, instead please use '%s", name, f.To)
+	}
+
+	forwardedNames[name] = f.To
+	fs.MarkDeprecated(name, f.Message) //nolint: errcheck
```

Comment:
can we `MarkDeprecated` similar to how we `MarkHidden`?  if/when we ever get around to fixing `errcheck` linting rules, it'll be easier to revert these things

Reply:
In particular this place, there is no call of `MarkDeprecated` needed now, I was trying to display the message, but seems like we name forwarding is sufficient here.